app/models/grda_warehouse/hud/project.rb

```diff
@@ -332,18 +335,31 @@ def serves_children?
     # End Standard Cohort Scopes
     #################################
 
-    scope :viewable_by, ->(user) do
+    # A single scope to determine if a user can access a project within a particular context
+    #
+    # @param user [User] user viewing the project
+    # @param project_scope [Symbol] a symbolized scope name that is merged into the vieable projects
+    #   within the context of reporting project_scope is almost always non_confidential
+    #   within the client dashboard context, project_scope is :all, which includes confidential projects
+    #   names of confidential projects are obfuscated unless the user can_view_confidential_project_names
+    scope :viewable_by, ->(user, project_scope: :non_confidential) do
       qc = ->(s) { connection.quote_column_name s }
       q  = ->(s) { connection.quote s }
 
-      where(
+      query = where(
         [
           has_access_to_project_through_viewable_entities(user, q, qc),
           has_access_to_project_through_organization(user, q, qc),
           has_access_to_project_through_data_source(user, q, qc),
           has_access_to_project_through_coc_codes(user, q, qc),
         ].join(' OR '),
       )
+
+      # If a user can't report on confidential projects, exclude them entirely
+      # return query if user.can_report_on_confidential_projects?
+      return query if user.can_report_on_confidential_projects?
+
+      query.send(project_scope)
```

Comment:
Hmm, I think if we're implementing it like this, the parameter should just be a flag, like `ignore_can_report_on_confidential_projects: false` (and we pass `true` when invoking from HUD reports or the Dashboard).

Otherwise the function signature seems misleading, because it's not clear that it is coupled to confidentiality specifically.

Other places where we pass a `scope` as a parameter like this, it gets merged with the resulting scope. Since we use that pattern elsewhere, I would expect `viewable_by(user, some_other_scope)` to return only projects in `some_other_scope`. But, that would only happen if my user has `can_report_on_confidential_projects`, which sounds like a tricky bug to find down the line.

Reply:
I think I'm following.  I don't want to pass a boolean because there are times that scope should be something other than `:all` or `:non_confidential` (at least I think there will be.  What if we name it `non_confidential_scope_limiter` or something like that?